tools/data/textdet/ilst_converter.py

```diff
@@ -0,0 +1,206 @@
+
+    Returns:
+        img_info (dict): The dict of the img and annotation information
+    """
+    assert isinstance(files, tuple)
+
+    img_file, gt_file = files
+    assert osp.basename(gt_file).split('.')[0] == osp.basename(img_file).split(
+        '.')[0]
+    # read imgs while ignoring orientations
+    img = mmcv.imread(img_file, 'unchanged')
+
+    img_info = dict(
+        file_name=osp.join(osp.basename(img_file)),
+        height=img.shape[0],
+        width=img.shape[1],
+        segm_file=osp.join(osp.basename(gt_file)))
+
+    if osp.splitext(gt_file)[1] == '.xml':
+        img_info = load_xml_info(gt_file, img_info)
+    else:
+        raise NotImplementedError
+
+    return img_info
+
+
+def load_xml_info(gt_file, img_info):
+    """Collect the annotation information.
+
+    Args:
+        gt_file (str): The path to ground-truth
+        img_info (dict): The dict of the img and annotation information
+
+    Returns:
+        img_info (dict): The dict of the img and annotation information
+    """
+    obj = ET.parse(gt_file)
+    root = obj.getroot()
+    anno_info = []
+    for object in root.iter('object'):
+        word = object.find('name').text
+        iscrowd = 1 if len(word) == 0 else 0
+        x1 = int(object.find('bndbox').find('xmin').text)
+        y1 = int(object.find('bndbox').find('ymin').text)
+        x2 = int(object.find('bndbox').find('xmax').text)
+        y2 = int(object.find('bndbox').find('ymax').text)
+
+        x = max(0, min(x1, x2))
+        y = max(0, min(y1, y2))
+        w, h = abs(x2 - x1), abs(y2 - y1)
+        bbox = [x1, y1, w, h]
+        segmentation = [x, y, x + w, y, x + w, y + h, x, y + h]
+        anno = dict(
+            iscrowd=iscrowd,
+            category_id=1,
+            bbox=bbox,
+            area=w * h,
+            segmentation=[segmentation])
+        anno_info.append(anno)
+
+    img_info.update(anno_info=anno_info)
+
+    return img_info
+
+
+def judge_latin(word):
+    for char in word:
+        if char not in string.ascii_letters:
+            return False
+    return True
+
+
+def split_train_test_list(full_list, test_ratio, shuffle=True):
+    """Split list by test_ratio
+
+    Args:
+        full_list (list): List to be splited
+        test_ratio (float): Ratio for test set
+        shuffle (bool): Whether to random sample test set
+
+    return:
+        list(list, list): train_list and test_list
+    """
+
+    n_total = len(full_list)
+    offset = int(n_total * test_ratio)
+    if n_total == 0 or offset < 1:
+        return [], full_list
+    if shuffle:
+        random.shuffle(full_list)
+    test_list = full_list[:offset]
+    train_list = full_list[offset:]
+    return [train_list, test_list]
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='Generate training and test set of ILST ')
+    parser.add_argument('root_path', help='Root dir path of ILST')
+    parser.add_argument(
+        '--preserve-vertical',
```

Comment:
We should keep consistency for the delimiter. here preserve-vertical uses '-' while others use '\_'. So let's use underscore '\_' for other later PRs.